lib/internal/bootstrap_node.js

```diff
@@ -123,7 +123,7 @@
         const internalModule = NativeModule.require('internal/module');
         internalModule.addBuiltinLibsToObject(global);
         evalScript('[eval]');
-      } else if (process.argv[1]) {
+      } else if (process.argv[1] && process.argv[1] !== '-') {
```

Comment:
IMHO `--` is much more common alias for "the rest will be piped throw `stdin`". If it's not too much work I'd be happy if you adjust.